components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OAS3Parser.java

```diff
@@ -1259,12 +1260,17 @@ public String getOASDefinitionWithTierContentAwareProperty(String oasDefinition,
      */
     private boolean isDefaultGiven(String swaggerContent) throws APIManagementException {
         OpenAPI openAPI = getOpenAPI(swaggerContent);
-        boolean isDefaultIsGiven = true;
-        SecurityScheme checkDefault = openAPI.getComponents().getSecuritySchemes().get(OPENAPI_SECURITY_SCHEMA_KEY);
-        if (checkDefault == null) {
-            isDefaultIsGiven = false;
+        boolean isDefaultGiven = true;
+        try {
+            SecurityScheme checkDefault = openAPI.getComponents().getSecuritySchemes().get(OPENAPI_SECURITY_SCHEMA_KEY);
+            if (checkDefault == null) {
+                isDefaultGiven = false;
+            }
+            return isDefaultGiven;
+        } catch (NullPointerException e) {
+            isDefaultGiven = false;
+            return isDefaultGiven;
         }
```

Comment:
Don't ever catch NPEs. We need to have pre-checks for null before executing a code that does an NPE.